tests/tracer/test_span.py

```diff
@@ -375,15 +375,49 @@ def test_span_links(self):
         }
         s1.link_span(s2.context, link_attributes)
 
-        assert s1._links == [
-            SpanLink(
-                trace_id=2,
-                span_id=1,
-                tracestate="dd=s:1,congo=t61rcWkgMzE",
-                flags=1,
-                attributes=link_attributes,
-            )
+        assert s1.get_link(s2.span_id) == SpanLink(
+            trace_id=s2.trace_id,
+            span_id=s2.span_id,
+            tracestate="dd=s:1,congo=t61rcWkgMzE",
+            flags=1,
+            attributes=link_attributes,
+        )
+
+    def test_init_with_span_links(self):
+        links = [
+            SpanLink(trace_id=1, span_id=10),
+            SpanLink(trace_id=1, span_id=20),
+            SpanLink(trace_id=2, span_id=30, flags=0),
+            SpanLink(trace_id=2, span_id=30, flags=1),
         ]
+        s = Span(name="test.span", links=links)
+
+        assert len(s._links) == 3
+        assert s.get_link(10) == links[0]
+        assert s.get_link(20) == links[1]
+        # duplicate links are overwritten (last one wins)
+        assert s.get_link(30) == links[3]
+
+    def test_set_span_link(self):
+        s = Span(name="test.span")
+        s.set_link(trace_id=1, span_id=10)
+        s.set_link(trace_id=1, span_id=20)
+        s.set_link(trace_id=2, span_id=30, flags=0)
+
+        with mock.patch("ddtrace.span.log") as log:
+            s.set_link(trace_id=2, span_id=30, flags=1)
+        log.debug.assert_called_once_with(
```

Comment:
Why don't we call this within the `with` bracket?

Reply:
The debug log is capture in the with block. The result is then stored in the log variable. We can nest `log.debug.assert_called_once_with` in the with block and the test should still pass. I am not sure what the best practice is for this 😓

Reply:
Yeah this is just a nit question, if it works then I also don't have strong opinions. I just assumed it would've failed if you tried to make the assertion on `log` outside of `with` context... but I guess not 😂